src/pages/storage/StorageBucketKeys.tsx

```diff
@@ -0,0 +1,295 @@
+import StorageBucketKeyActions from "./actions/StorageBucketKeyActions";
+import StorageBucketKeyBulkDelete from "./actions/StorageBucketKeyBulkDelete";
+
+const collapsedViewMaxWidth = 1400;
+export const figureCollapsedScreen = (): boolean =>
+  window.innerWidth <= collapsedViewMaxWidth;
+
+interface Props {
+  bucket: LxdStorageBucket;
+}
+
+const StorageBucketKeys: FC<Props> = ({ bucket }) => {
+  const docBaseLink = useDocs();
+  const [query, setQuery] = useState<string>("");
+  const notify = useNotify();
+  const [selectedNames, setSelectedNames] = useState<string[]>([]);
+  const [processingNames, setProcessingNames] = useState<string[]>([]);
+  const [isSmallScreen, setSmallScreen] = useState(figureCollapsedScreen());
+  const { project } = useCurrentProject();
+  const { data: keys = [] } = useBucketKeys(bucket, project?.name ?? "");
+
+  useEffect(() => {
+    const validNames = new Set(keys?.map((key) => key.name));
+    const validSelections = selectedNames.filter((name) =>
+      validNames.has(name),
+    );
+    if (validSelections.length !== selectedNames.length) {
+      setSelectedNames(validSelections);
+    }
+  }, [keys]);
+
+  const filteredKeys =
+    keys?.filter((key) => {
+      if (query) {
+        if (!key.name.toLowerCase().includes(query.toLowerCase())) {
+          return false;
+        }
+        if (!key.description.toLowerCase().includes(query.toLowerCase())) {
+          return false;
+        }
+        if (!key["access-key"].toLowerCase().includes(query.toLowerCase())) {
+          return false;
+        }
+        if (!key["secret-key"].toLowerCase().includes(query.toLowerCase())) {
+          return false;
+        }
+      }
+      return true;
+    }) ?? [];
+
+  const hasKeys = keys && keys.length > 0;
+
+  const headers = [
+    {
+      content: isSmallScreen ? (
+        <>
+          Name
+          <br />
+          <div className="header-second-row">Role</div>
+        </>
+      ) : (
+        "Name"
+      ),
+      sortKey: isSmallScreen ? "role" : "name",
+      className: "name",
+    },
+    {
+      content: "Role",
+      sortKey: "role",
+      className: "role",
+    },
+    ...(isSmallScreen
+      ? []
+      : [
+          {
+            content: "Description",
+            sortKey: "description",
+            className: "description",
+          },
+        ]),
+    { content: "Access key", className: "key-field", sortKey: "access-key" },
+    { content: "Secret key", className: "key-field", sortKey: "secret-key" },
+    { "aria-label": "Actions", className: "actions" },
+  ];
+
+  const rows = filteredKeys.map((key) => {
+    return {
+      key: key.name,
+      className: "u-row",
+      name: key.name,
+      columns: [
+        {
+          content: (
+            <>
+              <div className="u-truncate" title={`Key ${key.name}`}>
+                <ItemName item={key} />
+              </div>
+              {isSmallScreen && key.role}
```

Comment:
Can we add the muted text css class to the role, to make it easier to scan by name in the collapsed form?

Also, we probably shouldn't collapse in the mobile view with cards.

![image](https://github.com/user-attachments/assets/b4872f7d-e2fd-43ff-9967-ef4b43baebaa)

Reply:
The cards are fine, I think they should stay. My concern was on combining name and role when the card view is active. In the card, it is better to not combine them, but keep them as two different fields/columns. Because we don't have the problem with limited space for the columns there.

In the screen above we see "admin admin", one is the name, the other one the role. The "The admin key" description gets labelled as role wrongly. Better would be to have a heading "role" with value "admin" and another separate field with the name "admin".